test/extended/images/layers.go

```diff
@@ -0,0 +1,181 @@
+package images
+
+import (
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	kapi "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	buildapi "github.com/openshift/api/build/v1"
+	imageapi "github.com/openshift/api/image/v1"
+	buildclientset "github.com/openshift/client-go/build/clientset/versioned"
+	imageclientset "github.com/openshift/client-go/image/clientset/versioned"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[Feature:ImageLayers] Image layer subresource", func() {
+	defer g.GinkgoRecover()
+	var oc *exutil.CLI
+	var ns []string
+
+	g.AfterEach(func() {
```

Comment:
wait for builder service account in a beforeeach

Reply:
don't builds wait for builder service account?

Reply:
I don't try to access anything before builder service account is available.

Reply:
> don't builds wait for builder service account?

well, I should have said "wait for the builder service account's secrets".  (the waitforserviceaccount helper does both)

if the SA exists but the secrets don't, the build will fail (because it can't pull the base image or push the app image)  We do have some logic in the build controller now that waits for build secrets in certain cases:

https://github.com/openshift/origin/blob/master/pkg/build/controller/build/build_controller.go#L525-L543

given that i'm actually not sure why we saw some of the pull issues we saw earlier, unless those were only happening on deployments, not builds.